app/services/importer_service.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+class ImporterService
+  def initialize(data_set_id, fields: {})
+    @data_set_id = data_set_id
+    @fields = fields
+    @per_page = 500
+    @total_records = fetch_total_records_count.to_i
```

Comment:
Not sure about doing the `fetch_total_records_count` in the initialize block here rather than in the `import` call

Reply:
moved